kernel/cpu/src/aarch64.rs

```diff
@@ -50,20 +33,25 @@ pub fn current_cpu() -> CpuId {
     MpidrValue(MPIDR_EL1.get() as u64).into()
 }
 
-impl CpuId {
-    /// Reads an affinity level from this CpuId
+/// A unique identifier for a CPU core, read from the `MPIDR_EL1` register on aarch64.
+#[derive(
+    Clone, Copy, Debug, Display, PartialEq, Eq, PartialOrd, Ord,
+    Hash, Binary, Octal, LowerHex, UpperHex,
+)]
+#[repr(transparent)]
+pub struct MpidrValue(u64);
+
+impl MpidrValue {
+    /// Reads an affinity level from this `MpidrValue`.
     ///
-    /// Valid affinity levels are 0, 1, 2, 3
+    /// Panics if an affinity value Valid affinity levels are 0, 1, 2, 3. Panics i
     pub fn affinity(self, level: u8) -> u8 {
         assert!(level < 4, "Valid affinity levels are 0, 1, 2, 3");
-
         (self.0 >> (level * 8)) as u8
```

Comment:
This code is wrong, affinity level 3 is not at bits [24:31] in an MpidrValue, it's in bits [32:39]. This code is valid for a compact CpuId u32.

Reply:
ah, thanks for catching my mistake, that was accidental.

Reply:
I have undone this change now, but personally I believe that the `affinity` method should _only_ be implemented on `MpidrValue` since it is specific to aarch64 only, whereas `CpuId` should be kept arch-agnostic.

I"ll let you make that change in your next PR.

In this way, `CpuId` is analogous to `PteFlags`, `ApicId` is like `PteFlagsX86_64`, and `MpidrValue` is like `PteFlagsAarch64`.